src/jit/lsra.cpp

```diff
@@ -5686,12 +5684,37 @@ regNumber LinearScan::tryAllocateFreeReg(Interval* currentInterval, RefPosition*
             // the next ref, remember it.
             else if ((bestScore & UNASSIGNED) != 0 && intervalToUnassign != nullptr)
             {
+#ifdef _TARGET_ARM_
+                // We should update a double register if physRegToUnassign is part of the double register
+                if (intervalToUnassign->registerType == TYP_DOUBLE &&
+                    !genIsValidDoubleReg(physRegToUnassign->regNum))
+                    physRegToUnassign = findAnotherHalfRegRec(physRegToUnassign);
+#endif
                 updatePreviousInterval(physRegToUnassign, intervalToUnassign, intervalToUnassign->registerType);
+#ifdef _TARGET_ARM_
```

Comment:
@CarolEidt OK I understood well. Let me check and modify again with considering what you said. Thank you.